enhancements/microshift/microshift-apiserver-custom-certs.md

```diff
@@ -0,0 +1,221 @@
+---
+title: microshift-apiserver-custom-certs
+authors:
+  - "@eslutsky"
+reviewers:
+  - "@jerpeter, MicroShift contributor"
+  - "@pacevedom, MicroShift contributor"
+  - "@ggiguash, MicroShift contributor"
+  - "@benluddy, OpenShift API server team"
+  - "@standa, OpenShift Auth Team"
+
+approvers:
+  - "@jerpeter"
+api-approvers:
+  - N/A
+creation-date: 2021-01-18
+last-updated: 2023-01-18
+tracking-link:
+  - https://issues.redhat.com/browse/USHIFT-2101
+see-also:
+  - microshift-apiserver-certs.md
+replaces:
+  - N/A
+superseded-by:
+  - N/A
+---
+
+# MicroShift API server custom external certificates
+## Summary
+
+This enhancement extends the Microshift apiserver Certs to allow the user to
+provide additional custom certificates (e.g. signed by a 3rd party CA) for API server SSL handkshake and external authentication.
+
+
+> Anytime the document mentions API server, it refers to kube API server.
+
+
+## Motivation
+Currently, users of Microshift cannot provide their own generated certificates,
+Some customers have very strict requirements regarding TLS certs. There are frequently intermediate proxies which allow only TLS traffic with certs that are signed by organization owned CAs.
+
+### User Stories
+* As a MicroShift administrator, I want to be able to add  organization generated certificates. 
+  - each certificate may contain:
+    - Single Common Name containing The API server DNS/IPAddress or a wildcard entry (wildcard certificate).
```

Comment:
yes, custom  certificate may contain other FQDN , as long as it can be trusted and resolvable by the client running the `oc`  command.

Reply:
@ShaunaDiaz I don't understand your question. Can you restate it? What are you trying to understand about the SCN field?

Reply:
@dhellmann I'm wondering what the limits are on entries to this field and if anything about those need to be noted to users.

Reply:
OK, @eslutsky is describing here that MicroShift will work with certificates that have fully qualified domain names for hosts (`"myhost.example.com"`) or wildcard entries (`"*.example.com"`). The "API server" reference is there because these certificates are only used when accessing the API server, not workloads running on the cluster. The different types of entries are only relevant because MicroShift will generate different kubeconfig files based on which is used, but it does not validate the inputs in any way. Whether a name is "valid" depends on the context the client is in when it makes the request (which DNS server, what's in `/etc/hosts`, etc.) so the only way to know for sure that it works is for the client to make the request. For the docs, we should just explain that both kinds of names are allowed and talk about the kubeconfig files.